kong/runloop/balancer.lua

```diff
@@ -656,6 +657,9 @@ local create_hash = function(upstream)
       if type(identifier) == "table" then
         identifier = table_concat(identifier)
       end
+
+    elseif hash_on == "cookie" then
+      identifier = ngx.var["cookie_" .. upstream[cookie_field_name]]
```

Comment:
Oh great, I misunderstood and thought that a plugin would manage the generation or the cookie, but I think this is a lot better. Thanks for the quick feedback!

Reply:
@Tieske I'm having a little trouble finding out how to add request cookies when proxying. All of the documentation I'm finding is for the latter part, adding cookies to the response in the `header-filter` phase.
Also, do you have any opinions about adding the [resty-cookie library](https://github.com/cloudflare/lua-resty-cookie) as a dependency? I'm noticing it's quite popular for this kind of use-case, but I don't like adding dependencies if not necessary (and it might not be).

Reply:
I _think_ I have something that will work, but would like some early feedback on it:
```lua
local get_request_cookies = function()
  local cookies = ngx.header["Cookie"] or {}

  if type(cookies) == "string" then
    cookies = {cookies}
  end

  return cookies

end

local create_hash = function(upstream)
  -- ... existing logic
  elseif hash_on == "cookie" then
  -- If the cookie doesn't exist, create it
  identifier = ngx.var["cookie_" .. upstream[cookie_field_name]]

  if not identifier then
    identifier = utils.uuid()
    ngx.ctx.HASH_ON_COOKIE = identifier

    local cookies = get_request_cookies()
    table.insert(cookies, cookie_field_name .. "=" .. identifier)
    ngx.header["Cookie"] = table_concat(cookies, "; ")
  end
end
```